ci/test/docs-widgets/webhooks-datagen.test.js

```diff
@@ -0,0 +1,156 @@
+import { beforeEach, describe, it, expect, vi } from "vitest";
+import { JSDOM } from "jsdom";
+import fs from "fs";
+import path from "path";
+
+describe("Webhooks Data Generator", async () => {
+  let window, document;
+  let isGenerating = false;
+
+  beforeEach(() => {
+    const htmlPath = path.join(
+      __dirname,
+      "../../../doc/user/layouts/shortcodes/plugins/webhooks-datagen.html"
+    );
+    const htmlContent = fs.readFileSync(htmlPath, "utf-8");
+    const dom = new JSDOM(htmlContent, {
+      runScripts: "dangerously",
+      resources: "usable",
+      url: "http://localhost",
+    });
+
+    window = dom.window;
+    document = window.document;
+
+    // Wait for all the scripts and content to load
+    return new Promise((resolve) => {
+      document.addEventListener("DOMContentLoaded", resolve);
+    });
+  });
+
+  it("should initialize the DOM elements correctly", async () => {
+    const webhookURLInput = window.document.getElementById("webhookURL");
+    const authPasswordInput = window.document.getElementById("authPassword");
+    const jsonSchemaTextarea = window.document.getElementById("jsonSchema");
+    const startButton = window.document.getElementById("startButton");
+    const stopButton = window.document.getElementById("stopButton");
+
+    await new Promise((r) => setTimeout(r, 650));
```

Comment:
https://vitest.dev/api/vi.html#vi-waitfor should make this test more reliable and probably faster in many cases.

Reply:
Oh nice! Thanks for that suggestion Robin!